### PR TITLE
Fix: preserve BLE GPS speed in CSV export for classic box and add tests

### PR DESCRIPTION
**Pull request by mashazyu**
The classic box sends GPS speed via BLE (sensor title "gps"), which was stored in the DB with the same key as phone GPS speed, causing the phone speed to be silently overwritten during CSV export.

### Detailed Changes
- Rename BLE GPS sensor title from "gps" to "sensor_gps" so new recordings store BLE GPS data under distinct keys (sensor_gps_*)
- Fix CSV export (organizeSensorData, collectSensorTitles) to detect duplicate (title, attribute) rows per geolocation and store the second value under a sensor_<title> column instead of overwriting, preserving both values for existing tracks in the DB
- Skip "sensor_gps" in OpenSenseMap upload (same behavior as "gps")
- Add sensor_gps_* to sensorOrder and sensor utility lookups
- Add tests for duplicate-key handling in isar_utils

! currently when data is uploaded to osem, no gps sensor data is uploaded

### Testing
Can you please verify that:
- both gps sensor values are displayed on both home and track overview pages
- data export
    - in csv format both existing and new tracks should have new rows for lat, long and speed with "sensor_" prefix
    - data export in osem format should just have duplicate lines for gps data.
    

### Checklist before requesting a review

[x] I have performed a self-review of my code
[x] I have added or updated tests
[ ] I have added or updated relevant documentation

### CODE REVIEW

File: lib/utils/sensor_utils.dart [hunk 1]
```diff
@@ -152,10 +152,13 @@ String? getTitleFromSensorKey(String key, String? attribute) {
     case 'acceleration_z':
       return 'Acceleration Z';
     case 'gps_latitude':
+    case 'sensor_gps_latitude':
```
Comment (mashazyu) [inline lib/utils/sensor_utils.dart:155]: **suggestion:** I think it's ok for now, until custom titles for sensors are implemented.

**Review by felixerdy — APPROVED** [review]
